src/components/application_manager/src/hmi_state.cc

```diff
@@ -40,7 +40,7 @@ namespace application_manager {
 HmiState::HmiState(utils::SharedPtr<Application> app,
                    const ApplicationManager& app_mngr,
                    StateID state_id)
-    : app_(app)
+    : app_id_(app->app_id())
```

Comment:
@shoamano83 according to my comment above, propose to pass `app->hmi_app_id()` value directly, instead of passing the shared pointer

Reply:
Updated with commit 14270c9b11d5ac824f1d615d8b2f36277d573b53.